### ISSUE TITLE
Crash on OpenCL Acceleration activation

### ISSUE THREAD

**Issue by Jean-Marc68** [qgis/QGIS#59617]
### What is the bug or the crash?

QGis give message box "QGIS ended" unexpectedly" when I try to activate OpenCL acceleration

### Steps to reproduce the issue

In Preferences/Options/Aceleration, clic to activate OpenCL acceleration

## User Feedback

In Preferences/Options/Acceleration I cliqued to activate OpenCL acceleration.
Here a my computer specs :
Processor : i9-9900K CPU @ 3.60GHz
RAM : 32 Go
HDD : SSD 2 To
Graphic Card : NVIDIA Quadro P2200
Win11 v.26100.2454
QGIS 3.34.12

## Report Details

**Python Stack Trace**
```
Windows fatal exception: access violation

Current thread 0x00003ca4 (most recent call first):
  <no Python frame>
```

**Stack Trace**

```

cuGetExportTable :
cuGetExportTable :
clGetExportTable :
clGetExportTable :
clGetExportTable :
clGetExportTable :
clGetExportTable :
khrIcdVendorAdd :
khrIcdOsVendorsEnumerate :
RtlRunOnceExecuteOnce :
InitOnceExecuteOnce :
clGetPlatformIDs :
QgsOpenClUtils::init :
QgsOpenClUtils::init :
QgsOpenClUtils::init :
QgsOpenClUtils::errorText :
QgsOpenClUtils::activate :
QgsQtLocationConnection::startSatelliteMonitor :
QgsOpenClUtils::init :
QgsOpenClUtils::available :
QgsOptions::~QgsOptions :
QObject::qt_static_metacall :
QAbstractButton::setChecked :
QCheckBox::nextCheckState :
QAbstractButton::click :
QAbstractButton::mouseReleaseEvent :
QWidget::event :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QApplicationPrivate::sendMouseEvent :
QSizePolicy::QSizePolicy :
QSizePolicy::QSizePolicy :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QGuiApplicationPrivate::processMouseEvent :
QWindowSystemInterface::sendWindowSystemEvents :
QEventDispatcherWin32::processEvents :
qt_plugin_query_metadata :
QEventLoop::exec :
QDialog::exec :
QgisApp::showOptionsDialog :
QgisApp::options :
QObject::qt_static_metacall :
QAction::activate :
QMenu::actionGeometry :
QMenu::actionGeometry :
QMenu::mouseReleaseEvent :
QWidget::event :
QMenu::event :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QApplicationPrivate::sendMouseEvent :
QSizePolicy::QSizePolicy :
QSizePolicy::QSizePolicy :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QGuiApplicationPrivate::processMouseEvent :
QWindowSystemInterface::sendWindowSystemEvents :
QEventDispatcherWin32::processEvents :
qt_plugin_query_metadata :
QEventLoop::exec :
QCoreApplication::exec :
main :
BaseThreadInitThunk :
RtlUserThreadStart :
```

**QGIS Info**
QGIS Version: 3.34.12-Prizren
QGIS code revision: eeb6af3638
Compiled against Qt: 5.15.13
Running against Qt: 5.15.13
Compiled against GDAL: 3.9.3
Running against GDAL: 3.9.3

**System Info**
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 10.0.26100

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [ ] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by rouault** [comment]
Much likely not a QGIS bug but a wrong installation of OpenCL or buggy OpenCL drivers

**Comment by rouault** [comment]
@Jean-Marc68 Would you mind testing the build at https://github.com/qgis/QGIS/actions/runs/12057218622/artifacts/2246541251 and see if it avoids the crash?

**Comment by elpaso** [comment]
These two lines in the stacktrace look very wrong, memory corruption has already happended: 
QgsQtLocationConnection::startSatelliteMonitor :
QgsOpenClUtils::init :

**Comment by rouault** [comment]
My experience looking at various QGIS Windows stack traces is that some symbols mentionned in them are totally unrelated. This is likely due to optimisations omitting the frame pointer register which cause function name identification unreliable.

**Comment by Jean-Marc68** [comment]
I discovered something : That computer is a distant computer I access via Windows Remote Desktop and OpenCL acceleration works when I'm directly on the computer (not via remote desktop) but crashs QGIS if OpenCL is ON when connected via remote desktop.

**Comment by rouault** [comment]
@Jean-Marc68 did you get a chance of testing if https://github.com/qgis/QGIS/actions/runs/12057218622/artifacts/2246541251 avoids the crash?

**Comment by Jean-Marc68** [comment]
Yes. Qgis opens but acceleration is grayed with mention "QGIS is compiled with OpenCL support. GPU Acceleration is not available" (translated from french message). That message appears both on remote desktop and directly on computer.

**Comment by rouault** [comment]
> That message appears both on remote desktop and directly on computer.

hum, so there's a regression in the direct access, since OpenCL works with the official QGIS version? Do you get a message in the log messages pannel. There should be a message like "Unexpected exception of code {numeric_value_here} occurred while searching for OpenCL device. Note that the application may become unreliable and may need to be restarted." . What is the numeric value?

**Comment by Jean-Marc68** [comment]
Here is what's in the message panel (on the portable QGIS) 
Extensions : 
2024-11-28T12:08:26     INFO    DB Manager chargé (paquet db_manager)
2024-11-28T12:08:29     INFO    GRASS GIS Processing Provider chargé (paquet grassprovider)
2024-11-28T12:08:30     INFO    Processing chargé (paquet processing)
Processing : 
2024-11-28T12:08:29     CRITICAL    Problem with GRASS installation: GRASS was not found or is not correctly installed
Messages : 
2024-11-28T12:08:29     WARNING    Extension MetaSearch : Cette extension est désactivée car elle a précédemment planté QGIS.
General :
2024-11-28T12:08:30     INFO    QgsApplication state:
              - QGIS_PREFIX_PATH env var: 
              - Prefix: U:/QGIS/QGIS for Windows 64bit/qgis-portable-win64/QGIS-Portable
              - Plugin Path: U:/QGIS/QGIS for Windows 64bit/qgis-portable-win64/QGIS-Portable/lib/qgis/plugins
              - Package Data Path: U:/QGIS/QGIS for Windows 64bit/qgis-portable-win64/QGIS-Portable/share/qgis
              - Active Theme Name: default
              - Active Theme Path: U:/QGIS/QGIS for Windows 64bit/qgis-portable-win64/QGIS-Portable/share/qgis/resources/themes\default\icons/
              - Default Theme Path: :/images/themes/default/
              - SVG Search Paths: C:/PROGRA~1/QGIS33~1.12/apps/qgis-ltr/./svg/
              C:/Users/jmdeneyer/AppData/Roaming/QGIS/QGIS3\profiles\default/svg/
              U:/QGIS/QGIS for Windows 64bit/qgis-portable-win64/QGIS-Portable/share/qgis/svg/
              - User DB Path: U:/QGIS/QGIS for Windows 64bit/qgis-portable-win64/QGIS-Portable/share/qgis/resources/qgis.db
              - Auth DB Path: QSQLITE://C:/Users/jmdeneyer/AppData/Roaming/QGIS/QGIS3\profiles\default/qgis-auth.db

**Comment by Jean-Marc68** [comment]
And on the "regular" QGIS (via remote desktop) :
Extensions : 
2024-11-28T12:13:00     INFO    DB Manager chargé (paquet db_manager)
2024-11-28T12:13:01     INFO    GRASS GIS provider chargé (paquet grassprovider)
2024-11-28T12:13:03     INFO    Processing chargé (paquet processing)
General : 
             C:/Users/jmdeneyer/AppData/Roaming/QGIS/QGIS3\profiles\default/svg/
             Chemin de la BD utilisateur: C:/PROGRA~1/QGIS33~1.12/apps/qgis-ltr/./resources/qgis.db
             Chemin de la BD d'authentification: C:/Users/jmdeneyer/AppData/Roaming/QGIS/QGIS3\profiles\default/qgis-auth.db

**Comment by rouault** [comment]
@Jean-Marc68 Ah sorry I put you on the wrong track. The portable Windows build of the pull request is done using the mingw64 compiler whereas my fix/workaround is specific for builds done with Visual Studio C++... So no easy way to test it. I guess we'll need to merge it, and once it has landed into master, you'll be able to use the "qgis-dev" from the OSGeo4W network installer.

**Comment by Jean-Marc68** [comment]
Thanks. I'll wait. It's not a real problem to me. I was just bringing up an observation I made.

**Comment by agiudiceandrea** [comment]
@Jean-Marc68 , does the issue still occurs even after https://github.com/qgis/QGIS/pull/59618, running the latest QGIS versions?
